CI failure: "RuntimeError: tk.h version (8.6) doesn't match libtk.a version (8.5)"

<!-- Hi, and thank you for reporting an issue! Please take the time to first consider NeuroPoly's guidelines on issues (titles, descriptions, issue labels) before continuing:
https://www.neuro.polymtl.ca/software/contributing#opening_an_issue
-->


## Issue description


### Current behavior

CI is failing on {macOS} x {Python 3.7, Python 3.8}. Python3.6 and 3.9 are unaffected.

### Steps to reproduce

Run [CI](.github/workflows/run_tests.yml) on almost any branch currently and it will fail. Example failures:

* `master`: https://github.com/ivadomed/ivadomed/actions/runs/2340863632
* https://github.com/ivadomed/ivadomed/pull/1125: https://github.com/ivadomed/ivadomed/actions/runs/2295180486
  * Interestingly, this one only failed on 3.7, too
* https://github.com/ivadomed/ivadomed/pull/1126: https://github.com/ivadomed/ivadomed/actions/runs/2276043597
  * Interestingly, this one only failed on 3.7
* https://github.com/ivadomed/ivadomed/pull/1127: https://github.com/ivadomed/ivadomed/actions/runs/2321234595
* https://github.com/ivadomed/ivadomed/pull/1128: https://github.com/ivadomed/ivadomed/actions/runs/2327341106
* https://github.com/ivadomed/ivadomed/pull/1129: https://github.com/ivadomed/ivadomed/actions/runs/2340867212
* https://github.com/ivadomed/ivadomed/pull/1131: https://github.com/ivadomed/ivadomed/actions/runs/2341453544
 
Typical pattern of failures:

![Screenshot 2022-05-17 at 18-15-15 Update publish_package yml · ivadomed_ivadomed@c4c425b](https://user-images.githubusercontent.com/987487/168920668-24545ff3-9c20-4dca-b0ff-f4ac1a020704.png)


## Linked Issues

* Being tracked by GitHub staff at https://github.com/actions/setup-python/issues/402

The failures are all throwing the same [exception](https://github.com/ivadomed/ivadomed/runs/6478852629?check_suite_focus=true#step:7:382):

```
self = <tkinter.Tk object .>

    def _loadtk(self):
        self._tkloaded = 1
        global _default_root
        # Version sanity checks
        tk_version = self.tk.getvar('tk_version')
        if tk_version != _tkinter.TK_VERSION:
            raise RuntimeError("tk.h version (%s) doesn't match libtk.a version (%s)"
>                              % (_tkinter.TK_VERSION, tk_version))
E           RuntimeError: tk.h version (8.6) doesn't match libtk.a version (8.5)
```

I looked around for help and found an eclectic mixture of reports of this exact error, some on macOS, others in pyenv (which we don't use):

* https://github.com/jiansoung/issues-list/issues/15
* https://github.com/conda-forge/pillow-feedstock/pull/116
* https://github.com/pyenv/pyenv/issues/1375
* https://github.com/pyenv/pyenv/issues/2109
* https://github.com/pyenv/pyenv/issues/1545
* https://github.com/pyenv/pyenv/issues/1125
* https://github.com/spack/spack/issues/23780
* https://github.com/pyenv/pyenv/issues/94
* https://github.com/NixOS/nixpkgs/issues/10667
* https://github.com/Homebrew/legacy-homebrew/pull/41728

as well as this reported fix in `brew`:

* https://github.com/Homebrew/legacy-homebrew/issues/18657
* https://github.com/Homebrew/legacy-homebrew/pull/41728
* https://github.com/Homebrew/legacy-homebrew/issues/41132
* https://github.com/homebrew/homebrew-core/commit/4262fea34a709f430b0c685b15ab2773cbadf520

>  Fix RuntimeError: tk.h version (8.6) doesn't match libtk.a version (8.5)
>
> The problem was that CPPFLAGS and LDFLAGS for brewed Tk were specified to environment variable but Python (seems) prefer the CPPFLAGS in arguments specified to './configure', mean "-I#{MacOS.sdk_path}/usr/include" version is used.


The confusing thing is that that fix went in in 2015, so why is it showing up again now? Also are the Github runners even using brew's python? Do they have their own python build that accidentally recreated the bug? And why only 3.7 and 3.8?

--- a/.github/workflows/run_tests.yml
+++ b/.github/workflows/run_tests.yml
@@ -19,6 +19,11 @@ env:
   # (TravisCI quietly defined this on all their platforms, but we have to give it manually on GithubCI.)
   DEBIAN_FRONTEND: 'noninteractive'
   HDF5_USE_FILE_LOCKING: 'FALSE'
+  # Skip to the headless matplotlib renderer, which is less
+  # bug-prone in the constrained environment of CI
+  # Tip from a matplotlib dev: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3388#issuecomment-846091012
+  # Ref: https://matplotlib.org/stable/users/explain/backends.html
+  MPLBACKEND: 'Agg'
 
 jobs:
   ultra_matrix_test:

--- a/ivadomed/scripts/visualize_and_compare_testing_models.py
+++ b/ivadomed/scripts/visualize_and_compare_testing_models.py
@@ -9,6 +9,7 @@
 ###########################################################################################################
 
 import matplotlib
+from matplotlib import pyplot as plt
 import pandas as pd
 import numpy as np
 import itertools

--- a/ivadomed/scripts/visualize_and_compare_testing_models.py
+++ b/ivadomed/scripts/visualize_and_compare_testing_models.py
@@ -21,21 +21,7 @@ import argparse
 
 matplotlib.rcParams['toolbar'] = 'None'  # Remove buttons
 
-gui_env = ['TKAgg', 'GTKAgg', 'Qt4Agg', 'WXAgg']
-selected_gui_env = []
-for gui in gui_env:
-    try:
-        matplotlib.use(gui)
-        from matplotlib import pyplot as plt
-
-        selected_gui_env = gui
-        break
-    except:
-        continue
-# If none works
-if selected_gui_env == []:
-    from matplotlib import pyplot as plt
-
+if matplotlib.get_backend() == "agg":
     logger.warning("No backend can be used - Visualization will fail")
 else:
     logger.info(f"Using: {matplotlib.get_backend()}  gui")

--- a/ivadomed/utils.py
+++ b/ivadomed/utils.py
@@ -222,7 +222,6 @@ def plot_transformed_sample(before, after, list_title=None, fname_out="", cmap="
     if fname_out:
         plt.savefig(fname_out)
     else:
-        matplotlib.use('TkAgg')
         plt.show()
 
 

